Add support for 5 minute data from vnstat 2.x - compatible with vnstat 1.x

tomangert opened the pull request:
Tested with vnstat 2.3 and vnstat 1.14 on Ubuntu 16.04 / Apache 2.4.39

tomangert committed:
Add support for 5 minutes data

tomangert committed:
Fixed for json v1
Tested against vnstat 2.3 and vnstat 1.14

tomangert committed:
Make hour use 'log' scale
Use 'current' google chart version

tomangert committed:
Make graph wider
5min graph duration adjusts automatically to screen width

tomangert committed:
Make daily graph use log
Make graphs taller
More digits in graph data

tomangert committed:
Merge pull request #1 from tomangert/min5

Min5

tomangert committed:
Enable graph explorer
Clean up base calculation

tomangert committed:
Merge pull request #2 from tomangert/min5

Enable graph explorer

tomangert committed:
Fix vnstat v1 compatibility
Add zoom/pan instructions

tomangert committed:
Merge pull request #3 from tomangert/min5

Fix vnstat v1 compatibility

tomangert commented:
Add zoom and pan capability to 5min/hour/day graphs
More fixes / cleanup

alexandermarston commented:
@tomangert thanks for doing some work on this and getting it up to date. I'll have a look at this PR in detail tonight hopefully.

Does PR #36 need to be closed now this one is open?

tomangert commented:
Yes 36 can be closed - changes from 36 are included in 37.

alexandermarston commented:
A few comments with this one initially, @tomangert :

- The content now appears to take out the whole page width whereas before content was centered, could we please revert this to how it was?
- I'm not sure the use of being able to pan and zoom the graph, given the small dataset it is.
- Could you revert the formatting of the time in the hourly list to its original (i.e.: '8pm')

tomangert commented:
1) OK.
2) V2 allows you to configure the dataset size for each sampling interval. I have mine bumped up and it makes the pan/zoom pretty useful.
3) I can use am/pm instead of 24hr time - but if you have more than 24 hours in the data you need the day. I can check the time range and suppress the day if not needed.

alexandermarston commented:
@tomangert can you update your copy against the new changes and we can have a look what it looks like.

tomangert commented:
ok - probably tonight.  (gmt-4 here)

tomangert committed:
Merge main with:
    support for vnstat v2
    support for 5 minute data
    vnstat v1 compatible

    So far - graph in first tab is OK, other tabs are small and incomplete

tomangert committed:
Merge main with:
    support for vnstat v2
    support for 5 minute data
    vnstat v1 compatible

    So far - graph in first tab is OK, other tabs are small and incomplete

tomangert committed:
Fix directory name typo

Schokobecher commented:
Hey - any update on this?

tomangert committed:
Merge with changes from upstream.

Mostly working

tomangert committed:
Minimal change to add support for vnstat2 and it's 5 minute data. vnstat1 support is preserved (works with both)

tomangert committed:
Merge branch 'master' into min5

tomangert commented:
The history on this is a bit of a mess. The end result is 5 files with changes. Let me know if I should redo this in a new pull request.

alexandermarston commented:
@tomangert this looks good. Have you got an example of this running with the 5 minute dataset?

tomangert commented:
Example sent in an email.

The current merge request is the minimum changes to get something working.

Problems / enhancements left to deal with (~decreasing severity):
1) vnstat2 keeps a lot more data. Current code displays it all on the initial graph. but frequently it is too much to see the individual bars. It needs to set the initial zoom level to show about the same as vnstat1 - or set number of time periods in config file - or compute on the fly based on screen size.
2) Scroll-to-zoom and drag-to-pan help, but parameters need tweaking to allow more zoom range and finer control.
3) Need to add help text for scroll/drag

The rest are not directly 5 minute/vnstat2 related and may just be my preferences.

4) Logarithmic graph. This can be a config file option. My network is lightly loaded 90% of the time (~1% of bandwidth) Other times it is moderately loaded (interactive or streaming something) and occasionally there is something saturating the bandwidth (backup or big download or ...) . With a linear graph I can only see saturated vs next to nothing most of the time. A logarithmic graph shows a lot more nuance. 
5) Screen formatting. On medium to large screens the graphs are only using the middle half of the screen - about 1/4 of the screen is blank on both sides. For text that is useful to make it easier to read, but for graphs wider is better.

Commit:
Set initial zoom to 2H/1D/30D for 5min/hour/day
Allow wider range of zoom

Commit:
Fix vnstat2

Commit:
Don't sort top10

tomangert committed:
Merge pull request #5 from tomangert/enh

Fixs and enhancments

tomangert committed:
Add logarithmic graph option.
Fix occasional display error.
Clean up some floatingpoint math

tomangert committed:
Merge pull request #7 from tomangert/enh

Add logarithmic graph option.
